src/orion/client/experiment.py

```diff
@@ -877,11 +885,28 @@ def _verify_reservation(self, trial):
             raise RuntimeError(f"Reservation for trial {trial.id} has been lost.")
 
     def _maintain_reservation(self, trial):
-        self._pacemakers[trial.id] = TrialPacemaker(trial, self.storage)
+        self._pacemakers[trial.id] = TrialPacemaker(trial, self)
         self._pacemakers[trial.id].start()
 
+    def _update_heardbeat(self, trial):
```

Comment:
Why move this inside the experiment client instead of keeping it in the Pacemaker?

Reply:
The pacemaker was using the storage directly, but storage might not be instantiated for the service,
so the pacemaker need to be using the Client and the client will do what is right